File: Server/route/userContainer.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetUserContainerRoute(r *gin.Engine) {
	userContainer := r.Group("/userContainer")
	userContainer.Use(middleware.JWTAuthMiddleware())
	{
        userContainer.POST("GetContainerListByUser", api.GetContainerListByUser)
		userContainer.POST("CreateContainerByUser", api.CreateContainerByUser)
		userContainer.POST("DestroyContainerByUser", api.DestroyContainerByUser)
		userContainer.POST("DelayContainerByUser", api.DelayContainerByUser)
	}
}

File: Web/src/api/userContainer.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetContainerListByUser() {
  return apiClient.post("/userContainer/GetContainerListByUser").then(handleResponse).catch(handleError)
}

function CreateContainerByUser(data) {
  return apiClient.post("/userContainer/CreateContainerByUser", data).then(handleResponse).catch(handleError)
}

function DestroyContainerByUser(data) {
  return apiClient.post("/userContainer/DestroyContainerByUser", data).then(handleResponse).catch(handleError)
}

function DelayContainerByUser(data) {
  return apiClient.post("/userContainer/DelayContainerByUser", data).then(handleResponse).catch(handleError)
}

const userContainerApi = {
  GetContainerListByUser,
  CreateContainerByUser,
  DestroyContainerByUser,
  DelayContainerByUser
}

export default userContainerApi
